src/routes/transactions/entities/readable-descriptions/parsed-messages.ts

```diff
@@ -0,0 +1,111 @@
+import * as Messages from './index';
+import {
+  ContractMessages,
+  HumanReadableFragment,
+  ValueType,
+} from '../../mappers/common/readable-descriptions.mapper';
+
+type Expression = {
+  [key: string]: string;
+};
+
+const isValueType = (type: unknown): type is ValueType => {
+  return Object.values(ValueType).includes(type as ValueType);
+};
+
+/** This matches 2 patterns:
+ *  1. Match double curly braces consisting of 2 groups separated by a space
+ *  2. Match any non-whitespace character i.e. simple words
+ */
+const TEMPLATE_REGEX = /{{(.*?)\s(.*?)}}|(\S+)/g;
+
+// TODO: Write tests for this parser
+/**
+ * Go from a template string like this
+ * Send {{tokenValue 1}} to {{recipient 0}}
+ * to an array of HumanReadableFragment objects
+ * @param messages
+ */
+function parseMessages(messages: Expression): ContractMessages {
+  const contractMessages: ContractMessages = {};
+
+  for (const callSignature in messages) {
+    const template = messages[callSignature];
+
+    contractMessages[callSignature] = {
+      render: (to: string, params: unknown[]) => {
+        const fragments: HumanReadableFragment[] = [];
+
+        let match;
+
+        while ((match = TEMPLATE_REGEX.exec(template)) !== null) {
```

Comment:
You can use `String.replace(regex, (s, s1, s1) => { ... }` instead of a loop.

Reply:
String.replace would return a new string but we want to build an array of object depending on what is inside that string.